### PR TITLE
check if source files exist

### LINKED ISSUE #477: `dusty cp` gives very unhelpful error message when the file can't be found
For example:

```
/gc/gcchef/script @ 03:29 PM (sm-ssh)
-> dusty cp querysh:/gc/gc_state_game_buckets.csv /gc/gc_state_game_buckets.csv
Warning: Permanently added '[localhost]:2022' (RSA) to the list of known hosts.
rsync: link_stat "/cp/querysh/a860198c-3d3a-11e5-b4fd-a45e60d6b833" failed: No such file or directory (2)
rsync error: some files could not be transferred (code 23) at /SourceCache/rsync/rsync-45/rsync/main.c(1400) [receiver=2.6.9]
ERROR: Command 'rsync -e ssh -p 2022 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i /Users/scottmorse/.ssh/id_boot2docker -az --del --force --rsync-path sudo rsync --exclude */.git docker@localhost:/cp/querysh/a860198c-3d3a-11e5-b4fd-a45e60d6b833 /gc/gc_state_game_buckets.csv' returned non-zero exit status 23
```

My own fault for mistyping the filename, but took me a bit to figure out what I'd done wrong because of the error.

### PR DESCRIPTION
**Pull request by jsingle**
@thieman 

Resolves #477

### CODE REVIEW

File: dusty/systems/docker/files.py [hunk 1]
```diff
@@ -34,3 +34,8 @@ def move_file_inside_container(app_or_service_name, source_path, dest_path):
 
     _create_dir_in_container(client, container, parent_dir(dest_path))
     _move_in_container(client, container, source_path, dest_path)
+
+def container_path_exists(app_or_service_name, path):
+    client = get_docker_client()
```
Comment (thieman) [inline dusty/systems/docker/files.py]: We should see if we can memoize this if we haven't already. Probably a separate PR, though.

File: tests/integration/cli/cp_test.py [hunk 2]
```diff
@@ -89,3 +89,11 @@ def test_cp_dir_between_containers(self):
         self.run_command('cp busyboxa:/output busyboxb:/output')
         self.assertFileContentsInContainer('busyboxb', '/output/file1.txt', '1')
         self.assertFileContentsInContainer('busyboxb', '/output/file2.txt', '2')
+
+    def test_no_local_file(self):
+        output = self.run_command('cp {} busyboxa:/new-folder/file1.txt'.format('ghost-file'), raise_on_error=False)
+        self.assertInSameLine(output, 'ghost-file', 'does not exist')
```
Comment (thieman) [inline tests/integration/cli/cp_test.py]: Why wouldn't we check for the error instead?

File: dusty/systems/docker/files.py (hunk 1 above)
Comment (jsingle) [inline dusty/systems/docker/files.py]: good call

File: tests/integration/cli/cp_test.py (hunk 2 above)
Comment (jsingle) [inline tests/integration/cli/cp_test.py]: we can check for `CommandError` getting raised but that's pretty unspecific.

Comment (thieman) [inline tests/integration/cli/cp_test.py]: I guess it was already erroring before, we are just making the message nicer? If so I guess this makes sense

**Pull request merged**